adapters/beachfront/beachfront.go

```diff
@@ -111,427 +96,588 @@ type BeachfrontResponseSlot struct {
+			headers.Add("DNT", strconv.Itoa(int(*request.Device.DNT)))
 		}
 	}
-	return BannerEndpoint
-}
 
-func (a *BeachfrontAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
-	var beachfrontRequests BeachfrontRequests
-	var reqJSON []byte
-	var uri string
-	var errs = make([]error, 0, len(request.Imp))
-	var err error
-	var imps int
+	var reqCount = len(beachfrontRequests.ADMVideo) + len(beachfrontRequests.NurlVideo)
+	if len(beachfrontRequests.Banner.Slots) > 0 {
+		reqCount++
+	}
 
-	uri = getEndpoint(request)
+	var reqs = make([]*adapters.RequestData, reqCount)
 
-	beachfrontRequests, errs, imps = preprocess(request, uri)
+	var nurlBump = 0
+	var admBump = 0
 
-	if uri == VideoEndpoint {
-		reqJSON, err = json.Marshal(beachfrontRequests.Video)
-		uri = uri + beachfrontRequests.Video.AppId + VideoEndpointSuffix
-	} else {
-		/*
-			We will get here if request contains no Video imps, though it might have
-			Audio or Native imps as well as banner.
-		*/
-		reqJSON, err = json.Marshal(beachfrontRequests.Banner)
+	// At most, I only ever have one banner request, and it does not need the cookie, so doing it first.
+	if len(beachfrontRequests.Banner.Slots) > 0 {
+		bytes, err := json.Marshal(beachfrontRequests.Banner)
+
+		if err == nil {
+			reqs[0] = &adapters.RequestData{
+				Method:  "POST",
+				Uri:     bannerEndpoint,
+				Body:    bytes,
+				Headers: headers,
+			}
+
+			nurlBump++
+			admBump++
+		} else {
+			errs = append(errs, err)
+		}
 	}
 
-	if imps == 0 {
-		errs = append(errs, errors.New("no valid impressions were found"))
-		return nil, errs
+	if request.User != nil && request.User.BuyerUID != "" && reqCount > 0 {
+		headers.Add("Cookie", "__io_cid="+request.User.BuyerUID)
 	}
 
-	if err != nil {
-		errs = append(errs, err)
-		return nil, errs
+	for j := 0; j < len(beachfrontRequests.ADMVideo); j++ {
+		bytes, err := json.Marshal(beachfrontRequests.ADMVideo[j].Request)
+		if err == nil {
+			reqs[j+nurlBump] = &adapters.RequestData{
+				Method:  "POST",
+				Uri:     videoEndpoint + "=" + beachfrontRequests.ADMVideo[j].AppId,
+				Body:    bytes,
+				Headers: headers,
+			}
+
+			admBump++
+
+		} else {
+			errs = append(errs, err)
+		}
 	}
-	// ------------------------------------
 
-	headers := http.Header{}
-	headers.Add("Content-Type", "application/json;charset=utf-8")
-	headers.Add("Accept", "application/json")
+	for j := 0; j < len(beachfrontRequests.NurlVideo); j++ {
+		bytes, err := json.Marshal(beachfrontRequests.NurlVideo[j].Request)
 
-	if request.Device != nil {
-		addHeaderIfNonEmpty(headers, "User-Agent", request.Device.UA)
-		addHeaderIfNonEmpty(headers, "X-Forwarded-For", request.Device.IP)
-		addHeaderIfNonEmpty(headers, "Accept-Language", request.Device.Language)
-		if request.Device.DNT != nil {
-			addHeaderIfNonEmpty(headers, "DNT", strconv.Itoa(int(*request.Device.DNT)))
+		bytes = append([]byte(`{"isPrebid":true,`), bytes[1:]...)
```

Comment:
You might want to check the err first, otherwise this may panic on a nil bytes response. Consider moving this into the `if err == nil` clause.